src/components/TimeCardPage/Timesheet.tsx

```diff
@@ -0,0 +1,160 @@
+    const [startDate, setStartDate] = useState(moment().startOf('week').day(0)); 
+    const [endDate, setEndDate] = useState(moment().endOf('week').day(0)); 
+
+    const updateDateRange = (start, end) => {
+        // Callback for date-range picker - does any pre-processing when grabbing a new date 
+        setStartDate(start); 
+        setEndDate(end); 
+        setCurrentTimesheetsToDisplay (userTimesheets, start); 
+        //TODO - remove once we are finished setting up API calls for  this 
+        console.log("New date range has been selceted:\n\t %s \nto \n\t%s", start, end); 
+    }
+    
+    const [userTimesheets, setUserTimesheets] = useState([]); 
+    const [currentTimesheets, setCurrentTimesheets] = useState([]);
+    const [selectedTimesheet, setTimesheet] = useState(); 
+    const columns = defaultColumns 
+
+    //Pulls user timesheets, marking first returned as the active one
+    useEffect(() => {
+        // Uncomment this if you want the default one loaded 
+        //setTimesheet(testingTimesheetResp)
+        apiClient.getUserTimesheets().then(timesheets => {
+            setUserTimesheets(timesheets); 
+            //By Default just render / select the first timesheet for now 
+            setCurrentTimesheetsToDisplay (timesheets, startDate); 
+        });
+    }, [])
+
+    const processTimesheetChange = (timesheet) => {
+        //Adding the time entry to the table 
+        //apiClient.addTimeEntry(timesheet); 
+    }
+
+    const setCurrentTimesheetsToDisplay  = (timesheets, currentStartDate) => {
+        const newCurrentTimesheets  = timesheets.filter(sheet => moment.unix(sheet.StartDate).isSame(currentStartDate, 'day'));
+        newCurrentTimesheets.map(sheet => {
+                                if (sheet.TableData.length === 0){
+                                    sheet.TableData = defaultRows;
+                                } 
+                               return sheet;
+                            });
+
+        if (newCurrentTimesheets.length < 1){
+            newCurrentTimesheets.push(testingTimesheetResp); // TODO: change to make correct timesheets for the week
+        }
+
+        if (newCurrentTimesheets.length > 1){
+
+            const totalHoursForEachDay = {};
+
+            console.log(startDate);
+
+            // add the days in that stretch to dictionary 
+            // set all to 0
+            // iterate through each sheet and increment accordingly
+
+            for (let start = moment(startDate); start.add(7,'days'); start.add(1, 'days')){
+                totalHoursForEachDay[start.format("MM/DD/YY")] = 0;
+            }
+
+            console.log(totalHoursForEachDay);
+
+            newCurrentTimesheets.forEach(sheet => {
+                sheet.TableData.forEach(entry => {
+                    totalHoursForEachDay[moment.unix(entry.StartDate).set({'minutes':0, 'hours':0, 'seconds':0}).format("MM/DD/YY")] += Number(entry.Duration);
+                });
+            });
+
```

Comment:
For the new component, we need to make sure that from a currently selected list of timesheets that editing one will update the aggregations